### ISSUE TITLE
Data and editor behavior specification

### ISSUE THREAD

**Issue by amis92** [BSData/phalanx#7]
The general objective is to create a dataset (gamesystem + catalogues + rosters) that has a very strictly defined desired behavior (mostly in terms of roster editor behavior). Those expectations (desired behaviors) should be described as well.

I'd imagine it working like that:
A "scenario package" consists of a game system, a catalogue, a roster and a scenario (free-form/markdown). The scenario describes steps taken in a roster editor and expected changes/behavior (e.g. warnings, options being hidden) to be observed. The package could also include reference rosters that a user should get following the scenario steps.

The dataset (game system and catalogues) could be shared across all scenarios and include all necessary permutations of data to be checked in scenarios.

It's essentially a spec for BattleScribe as is, and we will be able to further enhance it in future as we make changes, making sure to keep a back-compat should we desire to do so.

It's a giant project overall (the spec, in full), but I think starting with a basic "sample dataset" should be quite straightforward.

**Comment by amis92** [comment]
Repo created at https://github.com/BSData/skw9k

**Comment by AlexMorman** [comment]
Do we have an estimated date of "completion" for the sample dataset skw90? There's no rush, but I was curious when I would be able to start writing out our expected behavior scenarios :) Thanks!

**Comment by amis92** [comment]
@AlexMorman I think it might take some time. Maybe we can parallelize work and you could start writing down some simple, generic scenarios based on an existing dataset you've used previously? The most basic things. It could speed up the overall progress, and maybe also inspire rylok and ghalidrim as to what is needed in our reference dataset.

**Comment by AlexMorman** [comment]
Good idea. I'll start to put together some basic scenarios over the weekend

**Comment by rahtolsGitHub** [comment]
Hi there, don't know If it's the right place....i think adding a search function for already entered data would be great.  At the moment, in the Data Editor, you get a listing of things already entered. If the list ist long you have to scroll  for ages and search for a certain entry manually. If doable a search box would be great (you enter text, the box proposing hits, bei it (shared) entries,.... ). 
Sorry if posting at the wrong place, didnt't know where else. 
Rahtol @ BSdata Devs, discord

**Comment by amis92** [comment]
@rahtolsGitHub It's not the best place, let me show a better one: https://github.com/BSData/phalanx/discussions/11 (there are similar do/don't like lists in the [Ideas discussions](https://github.com/BSData/phalanx/discussions/categories/ideas) for both Roster and Data editor.
